lib/pwwka/queue_resque_job_handler.rb

```diff
@@ -4,16 +4,26 @@
 module Pwwka
   # A handler that simply queues the payload into a Resque job.  This is useful
   # if the code that should respond to a message needs to be managed by Resque, e.g.
-  # for the purposes of retry or better failure management.
+  # for the purposes of retry or better failure management.  You can ask for the routing key and properties
+  # by setting `PWWKA_QUEUE_EXTENDED_INFO` to `true` in your environment.
   #
   # You should be able to use this directly from your handler configuration, e.g. for a Heroku-style `Procfile`:
   #
   #     my_handler: rake message_handler:receive HANDLER_KLASS=Pwwka::QueueResqueJobHandler JOB_KLASS=MyResqueJob QUEUE_NAME=my_queue ROUTING_KEY="my.key.completed"
+  #     my_handler_that_wants_more_info: rake message_handler:receive HANDLER_KLASS=Pwwka::QueueResqueJobHandler JOB_KLASS=MyOthgerResqueJob PWWKA_QUEUE_EXTENDED_INFO=true QUEUE_NAME=my_queue ROUTING_KEY="my.key.#"
   #
   # Note that this will not check the routing key, so you should be sure to specify the most precise ROUTING_KEY you can for handling the message.
   class QueueResqueJobHandler
     def self.handle!(delivery_info,properties,payload)
-      Resque.enqueue(ENV["JOB_KLASS"].constantize, payload)
+      args = [
+        ENV["JOB_KLASS"].constantize,
+        payload
+      ]
+      if ENV["PWWKA_QUEUE_EXTENDED_INFO"] == 'true'
```

Comment:
Instead of relying on an ENV variable, what about checking `ENV["JOB_KLASS"].constantize.method(:perform).parameters`. That route may be a bit more fragile however.

Reply:
That would be a way to make this forward compatible.  Let me investigate.